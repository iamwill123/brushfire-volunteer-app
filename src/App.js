import React, { useState, useEffect, useContext } from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import EventList from './components/EventList';
import VolunteerList from './components/VolunteerList';
import EventDetails from './components/EventDetails';
import { store } from './store';
// import { SortEventListSelector } from './components/SortEventListSelector';

import { ThemeProvider } from 'styled-components';
import { useDarkMode } from './styles/useDarkMode';
import { lightTheme, darkTheme } from './styles/theme';
import { GlobalStyles } from './styles/global';
import { Footer } from './components/Footer';
import { Header } from './components/Header';
import { Navbar } from './components/Navbar';

function App() {
  const { state } = useContext(store);
  const [loading, setLoading] = useState(true);
  const [eventList, setEventList] = useState(state);

  const [theme, toggleTheme, componentMounted] = useDarkMode();
  const themeMode = theme === 'light' ? lightTheme : darkTheme;

  // filter events by most recent as default
  // useEffect(() => {
  //   dispatch({ type: 'SORT_BY', sort: 'most recent' });
  // }, [dispatch]);

  useEffect(() => {
    setEventList(state);
    setLoading(false);
  }, [state]);

  if (!componentMounted || loading) {
    return <div>Loading...</div>;
  }

  return (
    <ThemeProvider theme={themeMode}>
      <>
        <GlobalStyles />
        <Router>
          <div>
            <Navbar theme={theme} />
            <Header theme={theme} toggleTheme={toggleTheme} />

            <Switch>
              <Route path="/event/:title" component={EventDetails} />
              <Route path="/volunteer" component={VolunteerList} />
              <Route path="/">
                <EventList eventList={eventList} />
              </Route>
              {/* <Route>
              <NoMatch />
            </Route> */}
            </Switch>
          </div>
        </Router>
        <Footer />
      </>
    </ThemeProvider>
  );
}

export default App;
