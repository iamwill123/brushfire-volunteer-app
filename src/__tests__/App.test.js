import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from '../App';

afterEach(cleanup);

// https://jestjs.io/docs/en/snapshot-testing
it('matches snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
