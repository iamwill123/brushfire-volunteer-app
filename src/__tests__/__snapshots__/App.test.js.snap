// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="nav-wrapper"
    >
      <ul
        class="nav"
      >
        <li>
          <a
            href="/"
            style="border-bottom: 0px; padding-bottom: 0px;"
          >
            <img
              alt="Brushfire fire logo"
              class="brushfire-logo"
              src="bf-logo-full-light-480.png"
              style="background: rgb(209, 65, 12);"
            />
          </a>
        </li>
        <ul>
          <li>
            <a
              aria-current="page"
              class="selectedLink"
              href="/"
            >
              Explore
            </a>
          </li>
          <li>
            <a
              href="/volunteer"
            >
              Your Events
            </a>
          </li>
        </ul>
      </ul>
    </div>
    <div
      class="header"
    >
      <img
        alt="volunteer logo"
        class="volunteer-logo"
        src="volunteer-logo.png"
      />
      <h1>
        Find your next volunteer opportunity
      </h1>
      <span>
        6 events near you
      </span>
      <div
        class="toggleModeWrapper"
      >
        <button
          class="sc-AxjAm bQtEZV"
        >
          <svg>
            sun.svg
          </svg>
          <svg>
            moon.svg
          </svg>
        </button>
      </div>
    </div>
    Loading...
  </div>
  <footer>
    <span>
      Brushfire Volunteer App
    </span>
    <br />
    <span>
      By: William Pei Yuan
    </span>
  </footer>
</DocumentFragment>
`;
